### PR TITLE
Preview Mode

### PR DESCRIPTION
**Pull request by kylehotchkiss**
Need special review on the Wordpress changes here because I'm not sure I put the pieces together correctly (for example, we didn't have an admin-env.php here to begin with)

### CODE REVIEW

**Comment by kylehotchkiss** [comment]
@ccorda this is deployed and working on bubs-next WP Engine staging server (https://bubsnexts.wpengine.com/wp-admin) now. Please give the code, especially the PHP portions some review. I realize this is a big chunk of code to be pulling in.

Once this is merged, we can enable preview mode for clients by:
1) making sure the URLs in admin-env are up to date
2) adding wordpress secret token and jwt signing secrets to wp-config of client sites (potentially can be streamlined but with no jwt signing token variable set, graphql will totally break on a site)
3) enabling jwt plugin
4) creating graphql user as editor
5) using CURL to get the auth token
6) saving auth token to Vercel

I estimate this adds about an hour or two to each site's setup. Please note that because the vercel staging urls change for every git branch, and the staging url needs to be hardcoded in php, actual preview mode for staging wordpress probably won't work as clients expect. Not sure if clients use staging enough for this to be an issue.

File: website/.eslintrc.json [hunk 1]
```diff
@@ -5,11 +5,10 @@
     "es6": true,
     "node": true
   },
-  "plugins": ["prettier", "react", "import"],
+  "plugins": ["react", "import"],
```
Comment (ccorda) [inline website/.eslintrc.json:8]: I'm open to a discussion about changing how we use prettier/eslint  -- but I think that should be it's own issue/PR, not rolled into this.

(FWIW, I think we're doing it in a non-conflicting way: https://github.com/prettier/eslint-plugin-prettier#recommended-configuration)

File: website/src/lib/wordpress.js [hunk 2]
```diff
@@ -90,36 +91,78 @@ let fragmentCategories = /* GraphQL */ `
...
 `;
 
-export async function getPreviewPost(id, idType = 'DATABASE_ID') {
-  const data = await fetchAPI(
-    `
-    query PreviewPost($id: ID!, $idType: PostIdType!) {
-      post(id: $id, idType: $idType) {
-        databaseId
-        slug
+export async function getContentTypes() {
```
Comment (ccorda) [inline website/src/lib/wordpress.js:94]: My original thought was that we'd add this to `getGlobalProps`, so we have it in our globals context. But if it's not useful anywhere other than here, that's not needed.

**Review by ccorda — COMMENTED** [review]
Small code review things from scanning. As I implement on another site I might have a few more.

File: website/src/lib/wordpress.js (hunk 2 above)
Comment (kylehotchkiss) [reply website/src/lib/wordpress.js:94]: I don't see a need for this anywhere else on the site and it actually requires authentication to get the rest path value. We don't want to add authenticated wordpress queries as a prereq for normal site functionality. The `uri` value we get from posts does everything this does, we just don't have access to that data from preview posts, which is what getContentTypes is doing for us.

**Comment by kylehotchkiss** [comment]
Merge conflicts resolved

**Pull request merged**